### PR TITLE
Refactor : set agent start with proxy if the switch set to -g (#23541)

### LINKED ISSUE #23541: Add agent start shell in proxy distribution start.sh
if start agent we need to add shell like '-javaagent:agent/shardingsphere-agent.jar' into the proxy start.sh manually.

this should be detected and startup automatically if the `-g` parameter is enabled.

### PR DESCRIPTION
**Pull request by taojintianxia**
Fixes #23541

Changes proposed in this pull request:
  - set agent start with proxy if the switch set to -g

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [x] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: distribution/proxy/src/main/resources/bin/start.sh [hunk 1]
```diff
@@ -134,6 +135,20 @@ if [ "$1" == "-v" ] || [ "$1" == "--version" ] ; then
...
+
+function enable_agent() {
+    AGENT_PARAM="";
+    AGENT_FILE="${DEPLOY_DIR}/agent/shardingsphere-agent.jar"
+    if [ -f "$AGENT_FILE" ]; then
+      AGENT_PARAM=" -javaagent:${AGENT_FILE} "
+    fi
+}
+
+if [ "$1" == "-g" ] || [ "$1" == "--agent" ] ; then
```
Comment (TeslaCN) [inline distribution/proxy/src/main/resources/bin/start.sh:147]: Is it possible that the `-g` is not in `$1`?

**Comment by taojintianxia** [comment]
image

File: distribution/proxy/src/main/resources/bin/start.sh (hunk 1 above)
Comment (taojintianxia) [reply distribution/proxy/src/main/resources/bin/start.sh]: > 
wow, thanks for your suggestion. I was doing that then.
this time it support random positon in parameters.

**Review by tristaZero — APPROVED** [review]

**Comment by TeslaCN** [comment]
> image

Have you fixed the problem in this screenshot?

**Pull request merged**